src/rpp/rpp/sources/defer.hpp

```diff
@@ -0,0 +1,50 @@
+//                   ReactivePlusPlus library
+//
+//           Copyright Aleksey Loginov 2023 - present.
+//  Distributed under the Boost Software License, Version 1.0.
+//     (See accompanying file LICENSE_1_0.txt or copy at
+//           https://www.boost.org/LICENSE_1_0.txt)
+//
+//  Project home: https://github.com/victimsnino/ReactivePlusPlus
+
+#pragma once
+
+#include <rpp/sources/fwd.hpp>
+#include <rpp/observables/observable.hpp>
+
+namespace rpp::details
+{
+template<constraint::decayed_type Type, typename Factory>
+struct defer_strategy
+{
+    using ValueType = Type;
+
+    Factory observable_factory;
+    void subscribe(auto&& obs) const { observable_factory().subscribe(std::forward<decltype(obs)>(obs)); }
+};
+}
+
+namespace rpp
+{
+template<constraint::decayed_type Type, typename Factory>
+using defer_observable = observable<Type, details::defer_strategy<Type, Factory>>;
+}
+
+namespace rpp::source
+{
+
+/**
+ * @brief Creates rpp::observable that calls the specified observable factory to create an observable for each new observer that subscribes.
+ *
+ * @tparam Factory the type of the observable factory
+ *
+ * @ingroup creational_operators
```

Comment:
Can you add doxygen example?

Reply:
Any idea how to generate a marble equivalent to https://reactivex.io/documentation/operators/defer.html in rxmarbles? Looks complicated no?

Reply:
Yeah, not sure if it is possible

Reply:
I think, simple doxygen example like this
https://github.com/victimsnino/ReactivePlusPlus/blob/v2/src/examples/rpp/doxygen/just.cpp
with something similar to new added test  would be enough

Reply:
done